.github/workflows/tests.yml

```diff
@@ -12,7 +12,7 @@ jobs:
 
   docs:
     name: build documentation
-    timeout-minutes: 10
+    timeout-minutes: 120
```

Comment:
This is a significant increase. Why is this necessary?

Reply:
One of the runs for the update-links branch took over an hour, so I thought I would bump up the limit a bit.

https://github.com/RaRe-Technologies/gensim/actions/runs/4454218032/jobs/7823263904

Reply:
I see now that the build from today only took a minute, so I'll drop the timeout bump commit.

https://github.com/RaRe-Technologies/gensim/actions/runs/4459623150/jobs/7832189964

Reply:
Done.

Reply:
I think the problem here is that the build _can_ take hours in theory. Some of the documentation examples take a really long time to build: that's why we cache their build output. So if somebody updates those examples, a rebuild will require a long time, and the build documentation action will time out.

We're kind of stuck between a rock and a hard place here.

1. We don't want to run these build steps for hours _all the time_ (e.g. for every commit)
2. We _do_ want to make it easier for people to contribute to documentation, which occasionally requires a long time to run

I think the best way to compromise between the above would be to move this build-and-stash-artifacts step to a separate action that people would trigger explicitly via a comment. We could set a long timeout (e.g. 2 hours, longer if necessary). We could restrict the range of people who can issue that trigger to e.g. project maintainers (if possible).

What do you think?

Reply:
Sounds like a reasonable plan.

Do we know how long the build takes when starting from scratch
after you delete all of the cache from the git repository?

I don't know GitHub workflows well enough to implement the plan myself.

So perhaps you can merge the existing docs fixes commits,
then modify the workflows to implement the plan?

Reply:
> So perhaps you can merge the existing docs fixes commits,
> then modify the workflows to implement the plan?

I'd rather modify the workflow first, then merge. That way, we won't break the existing workflow.